### DISCUSSION TITLE
Is it possible to implement a gdb-client with `gdbstub`?

### DISCUSSION THREAD

**Discussion by TD-Sky** [daniel5151/gdbstub#144] [Q&A] [answered]
Excuse me, I learned gdb remote serial protocol first time today so my question may seem silly.

I wanna construct an auto-test application with gdb and found `gdbstub` finally. After learning its examples I concluded that `gdbstub` is a library for setting stubs in the code so the emulator can serve as a gdbserver.

So, is it impossible to use `gdbstub` to implement a gdb client?

**Comment by daniel5151** [comment, marked as answer]
You are correct - `gdbstub` cannot be used to control a GDB client. It is only used to implement remote targets that a GDB client can connect to.

A cursory search shows that there are some other Rust crates out there that allow controlling a GDB client. I haven't used these myself, but they might be suitable for your project.

- https://github.com/dzfranklin/gdbmi
- https://github.com/anfedotoff/gdb-command

Cheers,
Daniel Prilik
